update page counter for both available sections

gituser789 opened the pull request:
This update will close #30

gituser789 commented:
The second update implements #31

hoelzer commented:
Thanks!

As written in the issue, I would remove your changes to the page counting and instead use the implementation https://github.com/hoelzer/dfg/pull/33 If you also agree, can you remove that part from the running PR so it's easier to merge?

Also, why using "E" as the default for own publications? Is there any reason behind that (although of course the user can easily change that). 

Also, I would suggest adding the function to both subsubsections:

```latex
%a maximum of ten publications
\subsubsection{Articles published by outlets with scientific quality assurance, book publications, and works accepted for publication but not yet published}
\newrefcontext[labelprefix=E]
\printbibliography[category=reviewed, heading=none, env=bibliographyNUM, resetnumbers]

\subsubsection{Other publications, both peer-reviewed and non-peer-reviewed}
% Own literature can be designated with a prefix. If the function is not desired, delete the "labelprefix" and "resetnumbers = false".
\newrefcontext[labelprefix=E]
\printbibliography[category=nonreviewed, heading=none, env=bibliographyNUM, resetnumbers=false]
```

Because then all own project-relevant publications, regardless of preprint or peer-reviewed would get the prefix.

gituser789 commented:
Hi,

i updated the pull request according your comments.
The `\newrefcontext[labelprefix=E]` is only needed for changes in the prefix, so it just appears twice: adding prefix for all own literatures, and in the last bibliograpy it is removed. In the english template, it was in the second bibliography instead of the first, by mistake. I also fixed this.

The "E" is for the german "Eigen", so i replaced the "E" in the english template to "O" for "Own". 

By the way, the english and the german template differ in `\printbibliography` at `env=bibliographyNUM`. The result is in both templates the same. Is there any reason for this?

hoelzer commented:
Hi, 

ah I see (I only checked the engl template so far), thanks for the explanation and fixes. 

> By the way, the english and the german template differ in \printbibliography at env=bibliographyNUM. The result is in both templates the same. Is there any reason for this?

Hm, good question. I will have a look but can't tell right now.

hoelzer commented:
@gituser789 I think we should add the `env=biliographyNUM` to the `dfg-german.tex` as well, otherwise I get something like this:

![image](https://user-images.githubusercontent.com/14393703/185170257-f74c4df1-2d45-4daa-a294-ccf3aa5be15a.png)

If I change the part in the `dfg-german.tex` according to the code in the engl template `dfg.tex`: 

```latex
\newrefcontext[labelprefix=E]
\printbibliography[category=reviewed, heading=none, env=bibliographyNUM, resetnumbers]

\subsubsection{Andere Veröffentlichungen mit und ohne
  wissenschaftliche Qualitätssicherung}
  \printbibliography[category=nonreviewed, heading=none, env=bibliographyNUM, resetnumbers=false]
```

I get:

![image](https://user-images.githubusercontent.com/14393703/185170758-2720cac6-beac-4a79-b11b-47ecbe1fd49b.png)

as expected. If you agree, can you change that as well please? 

Finally, how could I deactivate this feature? In the comment you wrote that one can just remove the `\newrefcontext[labelprefix=E]` and the `resetnumbers` parameter, but when I do it like that (in the engl template now):

```latex
% Own literature can be designated with a prefix. If the function is not desired, delete the "labelprefix" and "resetnumbers = false".
%\newrefcontext[labelprefix=O]
\printbibliography[category=reviewed, heading=none, env=bibliographyNUM]%, resetnumbers]

\subsubsection{Other publications, both peer-reviewed and non-peer-reviewed}
\printbibliography[category=nonreviewed, heading=none, env=bibliographyNUM]%, resetnumbers=false]
``` 

I get:

![image](https://user-images.githubusercontent.com/14393703/185171457-9dfcbecd-0e0c-4582-8a2e-01b317effb48.png)

And not counting from 1-4 as I would expect?

gituser789 commented:
Can you try all steps again deleting temporary files before? In case of no direct literature change, there might be some issues when modifying the display style, but not the literature. So force a full rebuild by deleting all temporary files (.aux, .bcf, ....).

The unset `env=...` variable was working for me as expected using the german template. This bibenvironment is set in `proposal.sty`, can not say exactly what it is for. Just redefining margins and labelwiths? Seems to has no direct interaction to mark own literature. Do you have more information? Then i will add this.

hoelzer commented:
> Can you try all steps again deleting temporary files before? In case of no direct literature change, there might be some issues when modifying the display style, but not the literature. So force a full rebuild by deleting all temporary files (.aux, .bcf, ....).

I'm using the `make` file in the repository which also cleans up intermediate files. So this should be fine. Sorry, I can't manage to deactivate the feature, maybe I'm doing something wrong? Currently I try:

```latex
\newrefcontext[labelprefix=]
\printbibliography[category=reviewed, heading=none]%, resetnumbers]

\subsubsection{Other publications, both peer-reviewed and non-peer-reviewed}
\printbibliography[category=nonreviewed, heading=none]%, resetnumbers=false]
```

in the engl template. but I get

![image](https://user-images.githubusercontent.com/14393703/185181750-5cd76af7-f39d-441e-8a20-2fa797832a7f.png)

---

> The unset env=... variable was working for me as expected using the german template. This bibenvironment is set in proposal.sty, can not say exactly what it is for. Just redefining margins and labelwiths? Seems to has no direct interaction to mark own literature. Do you have more information? Then i will add this.

Ah yeah you are right it's also defined in the `proposal.sty`. I also don't see any effect when removing it from the engl template. Unfortunately, I can't tell anymore what that is about. Maybe we can also just remove it from the engl template instead of adding unnecessary code to the german template.

hoelzer commented:
Ahhh I think I found it. I also need to comment out this part:

https://github.com/hoelzer/dfg/pull/32/files#diff-3de954a83932a308da4d92a694864a0df7632dc8762f65e362300e764855ab03R194

And then I get again:

![image](https://user-images.githubusercontent.com/14393703/185189686-0f3d73a3-3dc6-45cd-94dd-2ace278c5182.png)

like it was before.

hoelzer commented:
The `resetnumbers=false` can stay apparently...

```latex
\printbibliography[category=nonreviewed, heading=none, env=bibliographyNUM, resetnumbers=false]
```

and does, in my test, not impact the output. So It seems removing __both__ 

```latex
\newrefcontext[labelprefix=]
```

commands will restore the original behaviour of the template. Can you confirm that?

gituser789 commented:
Yes, i can confirm this.

Should i do any changes inside this PR?
adding the `env=...` variable?

Here is a comparison without and with the `env` variable:

without `env` variable
```
\subsubsection{Veröffentlichte Arbeiten aus Publikationsorganen mit
  wissenschaftlicher Qualitätssicherung, Buchveröffentlichungen sowie
  bereits zur Veröffentlichung angenommene, aber noch nicht
  veröffentlichte Arbeiten}

\newrefcontext[labelprefix=E]
\printbibliography[category=reviewed, heading=none]

\subsubsection{Andere Veröffentlichungen mit und ohne
  wissenschaftliche Qualitätssicherung}
\printbibliography[category=nonreviewed, heading=none]
```
![image](https://user-images.githubusercontent.com/62549000/185306608-5b97f5f2-02e9-43d4-998d-fa9e1fab97b0.png)

with `env` variable
```
\subsubsection{Veröffentlichte Arbeiten aus Publikationsorganen mit
  wissenschaftlicher Qualitätssicherung, Buchveröffentlichungen sowie
  bereits zur Veröffentlichung angenommene, aber noch nicht
  veröffentlichte Arbeiten}

\newrefcontext[labelprefix=E]
\printbibliography[category=reviewed, heading=none, env=bibliographyNUM]

\subsubsection{Andere Veröffentlichungen mit und ohne
  wissenschaftliche Qualitätssicherung}
\printbibliography[category=nonreviewed, heading=none, env=bibliographyNUM]
```
![image](https://user-images.githubusercontent.com/62549000/185306645-a9cb0d41-57b5-4fb7-aa37-e39486edd1dc.png)

The output seems to be identical. Can not see any differences. What do you guess?

hoelzer commented:
Yeah, for me it also seems identical. Can you please just remove the `env=` parameter from the engl template? If its not needed apparently why having it in at this position... 

And then I would merge this PR finally :) thx!

gituser789 commented:
The `env` variable was introduced about 6 month ago, can you have a look at this commit?
Please double check. Then i will remove the varialbe if necessary.

hoelzer commented:
Hey @gituser789 , I looked at the commit but can't exactly reconstruct why I added that `env=` parameter -.- 

Anyway, I would merge your changes now and then do another PR to 

* keep the `env=` and also add it to the german template (I decided now to better keep it)
* do some further polishing of the README.md

Thanks again for the PR and the help!

Cheers,
Martin
